identity-apps-core/apps/recovery-portal/src/main/webapp/includes/init-url.jsp

```diff
@@ -0,0 +1,172 @@
+<%--
+  ~ Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+  ~
+  ~ WSO2 LLC. licenses this file to you under the Apache License,
+  ~ Version 2.0 (the "License"); you may not use this file except
+  ~ in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~    http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+--%>
+
+<%@ page import="org.apache.commons.lang.StringUtils" %>
+<%@ page import="org.wso2.carbon.identity.core.util.IdentityTenantUtil" %>
+<%@ page import="org.wso2.carbon.identity.mgt.endpoint.util.IdentityManagementEndpointConstants" %>
+<%@ page import="org.wso2.carbon.identity.core.ServiceURLBuilder" %>
+<%
+    final String CONSOLE = "Console";
+    final String MY_ACCOUNT = "My Account";
+    
+    String TENANT_DOMAIN_KEY = "tenantDomain";
+    String TENANT_DOMAIN_SHORT = "t";
+    String USER_TENANT_DOMAIN_SHORT = "ut";
+    String SERVICE_PROVIDER_NAME_SHORT = "sp";
+    String SERVICE_PROVIDER_ID_SHORT = "spId";
+
+    String identityServerEndpointContextParam = application.getInitParameter("IdentityServerEndpointContextURL");
+    String samlssoURL = "../samlsso";
+    String commonauthURL = "../commonauth";
+    String oauth2AuthorizeURL = "../oauth2/authorize";
+    String oidcLogoutURL = "../oidc/logout";
+    String openidServerURL = "../openidserver";
+    String logincontextURL = "../logincontext";
+    String longwaitstatusURL = "/longwaitstatus";
+
+    String tenantDomain;
+    String userTenantDomain;
+    String tenantForTheming;
+    String userTenant;
+    String spAppName;
+    String spAppId;
+
+    if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+        tenantDomain = IdentityTenantUtil.resolveTenantDomain();
+        tenantForTheming = tenantDomain;
+        userTenant = tenantDomain;
+
+        spAppName = request.getParameter(SERVICE_PROVIDER_NAME_SHORT);
```

Comment:
Could u check whether we need to encode these params for security reasons?

Reply:
encoding these values everywhere it's read inside java code will not only add unnecessary extra processing but also might make the string follow an unintended encoding than where it is intended to be used. Shouldn't the approach be that the is value encoded before being used in the intended place?

Reply:
Agreed, but it also looks like the param value has not been encoded anywhere it is used in this file :(

Reply:
through the file `spAppName` is used only to either assign a value to it, to check if it's blank or if it equals to another string. Hence there is no security risk.